src/renderers/WebGLRenderer.js

```diff
@@ -1146,8 +1155,25 @@ function WebGLRenderer( parameters ) {
 
 		lights.length = 0;
 
-		opaqueObjectsLastIndex = - 1;
-		transparentObjectsLastIndex = - 1;
+		if ( scene._opaqueObjects === undefined ) {
```

Comment:
Instead of polluting the scene object I would have a internal object in `WebGLRenderer` like `renderLists[ scene.id ]`.

We used to pollute the scene graph like this. It's handy, but it becomes messy very quickly.